tests/unit/utils/aotf.spec.js

```diff
@@ -284,329 +280,359 @@ describe('aotf (Api On The Fly)', () => {
           ]
         }
       ]
+      const permissions = []
 
       // filter mutations from the context of the workflow
       const out1 = aotf.filterAssociations(
         aotf.cylcObjects.Workflow,
         aotf.tokenise('~a/b'),
-        mutations
+        mutations,
+        permissions
       )
       expect(
-        out1.map((item) => { return [item[0].name, item[1]] }).sort()
+        out1.map((item) => [item.mutation.name, item.requiresInfo]).sort()
       ).to.deep.equal([
-        ['a', false],
-        ['b', true],
-        ['c', true] // no cycle point in the context -> info required
+        ['argon', false],
+        ['boron', true],
+        ['carbon', true] // no cycle point in the context -> info required
       ])
 
       // filter mutations from the context of a cycle point
       const out2 = aotf.filterAssociations(
         aotf.cylcObjects.Workflow,
         aotf.tokenise('~a/b//c'),
-        mutations
+        mutations,
+        permissions
       )
       expect(
-        out2.map((item) => { return [item[0].name, item[1]] }).sort()
+        out2.map((item) => [item.mutation.name, item.requiresInfo]).sort()
       ).to.deep.equal([
-        ['a', false],
-        ['b', true],
-        ['c', false] // cycle point in the context -> no info is required
+        ['argon', false],
+        ['boron', true],
+        ['carbon', false] // cycle point in the context -> no info is required
       ])
     })
 
-    describe('iterateType', () => {
```

Comment:
The diff is not so helpful for the remainder of this file; you may want to tick "Hide whitespace"

<img src="https://user-images.githubusercontent.com/61982285/170094182-c7b42404-d32d-44aa-9629-184fdc6afb5e.png" width=250>